artichoke-backend/src/extn/core/time/args.rs

```diff
@@ -300,6 +347,112 @@ mod tests {
         assert_eq!(0, result.nanoseconds);
     }
 
+    #[test]
+    fn month_supports_string_values() {
+        let mut interp = interpreter();
+
+        let table = [
+            (b"[2022, 'jan']", 1),
+            (b"[2022, 'feb']", 2),
+            (b"[2022, 'mar']", 3),
+            (b"[2022, 'apr']", 4),
+            (b"[2022, 'may']", 5),
+            (b"[2022, 'jun']", 6),
+            (b"[2022, 'jul']", 7),
+            (b"[2022, 'aug']", 8),
+            (b"[2022, 'sep']", 9),
+            (b"[2022, 'oct']", 10),
+            (b"[2022, 'nov']", 11),
+            (b"[2022, 'dec']", 12),
+        ];
+
+        for (input, expected_month) in table {
+            let args = interp.eval(input).unwrap();
+            let mut ary_args: Vec<Value> = interp.try_convert_mut(args).unwrap();
+            let result: Args = interp.try_convert_mut(ary_args.as_mut_slice()).unwrap();
+
+            assert_eq!(expected_month, result.month);
+        }
+    }
+
+    #[test]
+    fn month_strings_are_case_insensitive() {
+        let mut interp = interpreter();
+
+        let table = [
+            (b"[2022, 'Feb']", 2),
+            (b"[2022, 'fEb']", 2),
+            (b"[2022, 'feB']", 2),
+            (b"[2022, 'FEb']", 2),
+            (b"[2022, 'FeB']", 2),
+            (b"[2022, 'fEB']", 2),
+            (b"[2022, 'FEB']", 2),
+        ];
+
+        for (input, expected_month) in table {
+            let args = interp.eval(input).unwrap();
+            let mut ary_args: Vec<Value> = interp.try_convert_mut(args).unwrap();
+            let result: Args = interp.try_convert_mut(ary_args.as_mut_slice()).unwrap();
+
+            assert_eq!(expected_month, result.month);
```

Comment:
it seems cheap to add an assert on `result.year` too if you think it might be valuable.

Reply:
IMO, this is good to add. You never know if someone tries `14` expecting it to wrap and removing other conditions etc!